### PR TITLE
Watermark

### PR DESCRIPTION
**Pull request by SRC-xiaojin**
## What is the purpose of the change

Reduce the times of eventcb callback

## Brief changelog

set read water mark to msgLen + 4,wait for receiving whole data; reset read water mark to 4 after user's callback

## Verifying this change

the body size of msg is 2k
msg count is 1W
use Producer.c and PullConsumer.cpp for test
As I tested, could reduce latency a litter

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: src/transport/TcpTransport.cpp [hunk 1]
```diff
@@ -277,6 +281,10 @@ void TcpTransport::readNextMessageIntCallback(BufferEvent* event, TcpTransport*
 
       transport->messageReceived(msg, event->getPeerAddrPort());
     }
+    /**
+     * reset read water mark to 4
+     */
+    event->setWatermark(EV_READ, 4, 0);
```
Comment (ifplusor) [inline src/transport/TcpTransport.cpp:287]: Move to line 259.

**Review by ifplusor — CHANGES_REQUESTED** [review]

File: src/transport/TcpTransport.cpp (hunk 1 above)
Comment (SRC-xiaojin) [reply src/transport/TcpTransport.cpp:287]: at line 259, the water matk is already 4
so I think need reset read water mark to 4 after received whole data and user's callback

Comment (ifplusor) [reply src/transport/TcpTransport.cpp:287]: If `needed` is greater than 0, `hdr` is incomplete, return and wait for more data. The function does not return here.

**Review by ifplusor — APPROVED** [review]
LGTM

**Pull request merged**